app/controllers/katello/api/v2/systems_controller.rb

```diff
@@ -139,6 +139,15 @@ def create
     param :host_collection_ids, Array, :desc => N_("Specify the host collections as an array")
     def update
       system_params = system_params(params)
+
+      if params[:facts] && params[:facts]['network.hostname']
+        host = Host.where(:name => params[:facts]['network.hostname'], :organization_id => @system.organization.id).first
+      elsif @system.facts['network.hostname']
+        host = Host.where(:name => @system.facts['network.hostname'], :organization_id => @system.organization.id).first
+      else
+        host = Host.where(:name => @system.name, :organization_id => @system.organization.id).first
+      end
+      system_params.merge!(:host_id => host.try(:id))
```

Comment:
Removed this as it was outside the scope of the PR (but useful for testing). Functionality won't be needed after host unification anyway.